atc/db/resource.go

```diff
@@ -936,12 +927,40 @@ func scanResource(r *resource, row scannable) error {
 			PipelineInstanceVars: r.pipelineInstanceVars,
 		}
 
-		if build.startTime.Valid {
-			r.buildSummary.StartTime = build.startTime.Time.Unix()
+		err := populateBuildSummary(r.buildSummary, lastCheckStartTime, lastCheckEndTime, lastCheckSucceeded, lastCheckBuildPlan)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func populateBuildSummary(buildSummary *atc.BuildSummary,
+	lastCheckStartTime, lastCheckEndTime pq.NullTime,
+	lastCheckSucceeded sql.NullBool,
+	lastCheckBuildPlan sql.NullString) error {
+	if lastCheckStartTime.Valid {
+		buildSummary.StartTime = lastCheckStartTime.Time.Unix()
+
+		if lastCheckEndTime.Valid && lastCheckStartTime.Time.Before(lastCheckEndTime.Time) {
+			buildSummary.EndTime = lastCheckEndTime.Time.Unix()
+			if lastCheckSucceeded.Valid && lastCheckSucceeded.Bool {
+				buildSummary.Status = atc.StatusSucceeded
+			} else {
+				buildSummary.Status = atc.StatusFailed
+			}
+		} else {
+			buildSummary.Status = atc.StatusStarted
 		}
+	} else {
+		buildSummary.Status = atc.StatusPending
```

Comment:
Yes, you are right. Before `populateBuildSummary` is called, we have checked `if lastCheckBuildId.Valid`, so this `else` cluster should never be reached. I'll change this clause to return an error.

Reply:
Do we have another way of being able to show that a build status is in `pending` state?

Reply:
@clarafu I think check build no longer has "pending" state since #6845.

Reply:
Oh interesting, I didn't know we had that issue. Thanks for pointing it out!

Reply:
Minor thing, but fwiw #6845 didn't change that behaviour - it just did it in a single query rather than splitting it up into two queries (that ran in the same transaction). It's been like this since checks as builds